packages/graphql-codegen-cli/src/codegen.ts

```diff
@@ -1,587 +1,483 @@
+      - Introspection JSON File
+      - URL of GraphQL endpoint
+      - Multiple files with type definitions
+      - String in config file
 
-        Unknown codegen template: "${template}"
+    Try to use one of above options and run codegen again.
 
-        Please make sure it's installed using npm or yarn.
+  `
+  );
+};
 
-          $ yarn add ${template} -D
+async function mergeSchemas(schemas: GraphQLSchema[]): Promise<GraphQLSchema> {
+  if (schemas.length === 0) {
+    return null;
+  } else if (schemas.length === 1) {
+    return schemas[0];
+  } else {
+    return remoteMergeSchemas({ schemas: schemas.filter(s => s) });
+  }
+}
 
-          OR
+export async function executeCodegen(config: Types.Config): Promise<FileOutput[]> {
+  function wrapTask(task: () => void | Promise<void>, source?: string) {
+    return async () => {
+      try {
+        await task();
+      } catch (error) {
+        if (source) {
+          error.source = source;
+        }
 
-          $ npm install ${template} -D
+        throw error;
+      }
+    };
+  }
 
-        Template should match package's name.
+  const result: FileOutput[] = [];
+  const commonListrOptions = {
+    exitOnError: true
+  };
+  const verboseOptions = {
+    ...commonListrOptions,
+    renderer: 'verbose',
+    nonTTYRenderer: 'verbose'
+  };
+  const listrOptions: any = {
+    ...commonListrOptions,
+    renderer: config.silent ? 'silent' : Renderer,
+    nonTTYRenderer: config.silent ? 'silent' : 'default',
+    collapse: true,
+    clearOutput: false
+  };
+  const listr = new Listr(process.env.VERBOSE || process.env.NODE_ENV === 'test' ? verboseOptions : listrOptions);
 
-      `);
-    }
-  }
+  let rootConfig: {
+    [key: string]: any;
+  } = {};
+  let schemas: Types.Schema[];
+  let documents: Types.OperationDocument[];
+  let generates: { [filename: string]: Types.ConfiguredOutput } = {};
+  let rootSchema: GraphQLSchema;
+  let rootDocuments: DocumentFile[] = [];
 
-  debugLog(`[executeWithOptions] using project: ${project}`);
+  function normalize() {
+    /* Load Require extensions */
+    const requireExtensions = normalizeInstanceOrArray<string>(config.require);
+    requireExtensions.forEach(mod => require(mod));
 
-  const configPath = path.resolve(process.cwd(), gqlGenConfigFilePath);
-  let config: GqlGenConfig = null;
+    /* Root templates-config */
+    rootConfig = config.config || {};
 
-  if (fs.existsSync(configPath)) {
-    getLogger().info(`Loading config file from: ${configPath}`);
-    config = JSON.parse(fs.readFileSync(configPath).toString()) as GqlGenConfig;
-    debugLog(`[executeWithOptions] Got project config JSON: ${JSON.stringify(config)}`);
-  }
+    /* Normalize root "schema" field */
+    schemas = normalizeInstanceOrArray<Types.Schema>(config.schema);
 
-  if (project && project !== '') {
-    getSpinner().log(`Using project: ${project}`);
-    if (config === null) {
+    if (schemas.length === 0) {
```

Comment:
@dotansimha Why do we expect `codegen.yml` to have global schemas? In 🇫🇷 🇩🇪 🔌 project, for example, we won't have one schema for all modules but one for each.

Reply:
#832